modules/mauth-proxy/README.md

```diff
@@ -5,31 +5,106 @@ The goal of this simple service is to append valid mAuth headers to any incoming
+            buffer_size_in_bytes: 524288
+            buffer_size_in_bytes: ${?MAUTH_PROXY_BUFFER_SIZE_IN_BYTES}
+        }
+
+        app {
+            uuid: ${?APP_UUID}
+            private_key: ${?APP_PRIVATE_KEY}
+        }
 
 ## Start the proxy with Docker
 
 The reason for doing this is so you can use UI REST clients such as [Postman](https://www.getpostman.com/postman) or [Paw](https://paw.cloud/) or even just plain old `curl` to hit Medidata resources
 
-1. Install docker
-  1. `brew install docker`
-1. Start docker host(`docker-machine start <host_name>` or this may already be running if you are using "Docker for Mac/Windows")  
-1. Check out this repository  
-1. cd to the root of this repository
-1. `export APP_UUID='<the mauth id of the app you are proxying to>`
-1. `export APP_PRIVATE_KEY='<the actual key (not the path) of the app you are proxying to>`
-1. `docker build --rm -t mauth-proxy .`
-1. `docker run --env APP_UUID --env APP_PRIVATE_KEY -it --rm -p 9090:9090 --name <your_name> mauth-proxy`
+### Installing Docker
+
+You need to have Docker installed on your machine in order to use Mauth Proxy. The simplest way to get Docker running 
+on a Mac or Windows machine is to follow the instructions at 
+<ahref="https://docs.docker.com/#/docker-for-mac"> Docker for Mac </a>.
+
+> Please note that `boot2docker` has been deprecated in favor of
+> `Docker for *` which is maintained by the Docker team.
+
+*Note that Mauth Proxy currently cannot be run on native Docker For Mac.
+In order to run mauth-proxy you should kill Docker native and start
+docker-machine.  See below.*
+
+You can create a default VM for docker by running the following
+command, which allocates sufficient space to run mauth-proxy:
+
+`docker-machine create --driver virtualbox default`
+
+Before starting docker-machine you need to make sure that docker
+native is not running.  If Native Docker is running you will see an
+icon in the System Menubar of a whale with a large crate on its back.
+To stop Native Docker click on the Whale and select "Quit" at the
+bottom of the menu.
+
+You can start docker-machine by running `docker-machine start`
+
+You will need to stop Native Docker and start docker-machine everytime
+you reboot your machine.
+
+Next you need to run `eval $(docker-machine env)` in your terminal in
+order to set up the terminal to run docker.  After doing this you
+should be able to type 'docker ps' and get a listing of all of the
+docker process running on your machine. Of course, at this point,
+that should be an empty list.  But if this runs without error then
+docker is ready to go.
+
+Please note again that Mauth Proxy does not currently work with Docker
+Native.  It is OK to have it installed on your machine, but in order
+to use mauth-proxy you need to have docker-machine running and you
+should turn off Native Docker.
+
+### Installing and Configuring AWS CLI
+
+We are using <a href="https://aws.amazon.com/ecr/">AWS ECR </a> as our
```

Comment:
More HTML links in this paragraph. I'd usually expect to see markdown-style links in markdown files.